opds_import.py

```diff
@@ -266,57 +247,31 @@ def import_from_feed(self, feed, even_if_no_author=False,
             except Exception, e:
                 # Rather than scratch the whole import, treat this as a failure that only applies
                 # to this item.
-                message = StatusMessage(500, "Local exception during import:\n%s" % traceback.format_exc())
-                status_messages[key] = message
-                # go into circulation_obs, and remove any on the same identifier, 
-                # so we don't try to make pools and works from it
-                if key in circulation_objs:
-                    del circulation_objs[key]
+                identifier, ignore = Identifier.parse_urn(self._db, key)
+                data_source = DataSource.lookup(self._db, self.data_source_name)
+                failure = CoverageFailure(identifier, traceback.format_exc(), data_source=data_source, transient=False)
```

Comment:
I've replaced StatusMessage with CoverageFailure, but CoverageFailure didn't have status codes, so I took them out. Would it be better to add an optional status code to CoverageFailure, even though it's not in CoverageRecord?